spinoso-string/src/impls.rs

```diff
@@ -9,6 +9,18 @@ use std::io;
 
 use crate::String;
 
+impl Default for String {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Clone for String {
+    fn clone(&self) -> Self {
+        Self::with_bytes_and_encoding(self.inner.as_vec().clone(), self.encoding())
+    }
+}
```

Comment:
Why can't we `#[derive(Clone)]`?

Reply:
:thinking: I think I did this because rust was complaining that `EncodedString` wasn't implementing Clone. However since the encoded types support Clone, that means EncodedString can support Clone, so I'll just do that instead! :+1: